lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -155,28 +332,82 @@ Function Download-File($module, $url, $dest, $headers, $credentials, $timeout, $
         }
     }
 
-    if (-not $module.CheckMode) {
-        # FIXME: Single-out catched exceptions with more specific error messages
-        Try {
-            $extWebClient.DownloadFile($url, $dest)
-        } Catch [System.Net.WebException] {
-            $module.Result.status_code = [int] $_.Exception.Response.StatusCode
-            $module.Result.elapsed = ((Get-Date) - $module_start).TotalSeconds
-            $module.FailJson("Error downloading '$url' to '$dest': $($_.Exception.Message)", $_)
-        } Catch {
-            $module.Result.elapsed = ((Get-Date) - $module_start).TotalSeconds
-            $module.FailJson("Unknown error downloading '$url' to '$dest': $($_.Exception.Message)", $_)
+    Try {
+
+        $tmpDest = [System.IO.Path]::GetTempFileName()
```

Comment:
We should be using the temp directory as set by Ansible. You can get this directory by running `Join-Path -Path $module.Tmpdir -ChildPath ([System.IO.Path]::GetRandomFileName())`.

The `Tmpdir` property of AnsibleModule will automatically create the temp directory when called and clean itself up on exit. Now people can specify their own custom temp directories through Ansible in a universal way.

Reply:
Fixed